src/hooks/lottery/v2/useRoundEndCountdown.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react'
+
+const useRoundEndCountdown = (endTime: number) => {
+  const [secondsRemaining, setSecondsRemaining] = useState(null)
+  const [startingSeconds, setStartingSeconds] = useState(Math.floor(Date.now() / 1000))
+
+  useEffect(() => {
+    const secondsRemainingCalc = endTime - startingSeconds
+    setSecondsRemaining(secondsRemainingCalc)
+
+    const tick = () => {
+      setSecondsRemaining((prevSeconds) => prevSeconds - 1)
+    }
+    const timerInterval = setInterval(() => tick(), 1000)
+    return () => clearInterval(timerInterval)
```

Comment:
you also need to clear it every time the effect run since you have a non empty deps array

Reply:
I think I've caught where this could be a problem, when the `nextEventTime` (formerly `endTime`) changes, the effect will re-run and a new timer will be created, [fixed here by clearing the timeout when the timer runs out](https://github.com/pancakeswap/pancake-frontend/blob/463c232aa80570bb09de2f55240d64e0985a0d8b/src/hooks/lottery/v2/useNextEventCountdown.ts#L16). Other than that, the useEffect won't re-run.

Reply:
@ChefHutch I'm not sure if it works as it looks in the snippet you shared. At least it reads a bit weird:
First time it runs `const timerInterval = setInterval(() => tick(), 1000)` the `timerInterval` inside `tick()` would be undefined I presume (which is fine). Next time what would be `timerInterval`? I think it will be undefined again since it is not declared before `tick()` is initialized. I don't think it will somehow remember `timerInterval` from previous useEffect run 🤔  
I think usually you track the timer id with refs or just another state variable.

Reply:
Yea you could be right Cheems - am I right in thinking that the impact of the issue you're describing is just that when it comes to `clearInterval(timerInterval)` - it will be calling it on undefined, thus clearing no interval?

It's possible that it is in scope and isn't a problem - but you're right managing the timer via a ref is way nicer,  I've updated that.

Reply:
Yeah there is no real issue I guess other than memory clogging + taking CPU time.